src/Extensions/AzureBlobPayloads/PayloadStore/IPayloadStore.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.DurableTask;
+
+/// <summary>
+/// Abstraction for storing and retrieving large payloads out-of-band.
+/// </summary>
+public interface IPayloadStore
```

Comment:
Super nit: I would classify this more as an abstract class than an interface. It's not something you would think would neat multiple inheritance and might have some base behavior in the future (like with the token stuff). 

In general, you can think of an interface as a label which you can apply to classes. `IDisposable`, it can be disposed. `IComparable` it can be compared. In this case you wouldn't really consider things as 'It can be a PayloadStore' and also other things. 

Anyway, that's a personal preference on how things look, feel free to ignore me :)